src/popup/Popup.tsx

```diff
@@ -1,23 +1,56 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import { getCurrentTabUrl } from '../utils';
 
 export default function Popup(): JSX.Element {
   const [url, setUrl] = useState<string>('');
 
+  const [autolikeEnabled, setAutolikeEnabled] = useState(null);
+
   useEffect(() => {
     getCurrentTabUrl((url) => {
       setUrl(url || 'undefined');
     });
+    chrome.storage.sync.get(['autoAwesome'], (result) => {
+      setAutolikeEnabled(result['autoAwesome']);
+    });
   }, []);
 
   const sendAMessage = () => {
     // Example of how to send a message to eventPage.ts.
     chrome.runtime.sendMessage({ popupMounted: true });
   };
 
+  const changeAutolike = useCallback(() => {
+    const currentLikeStatus = !autolikeEnabled;
+    setAutolikeEnabled((autolikeEnabled) => !autolikeEnabled);
+    const message = currentLikeStatus
+      ? 'enableAutoAwesome'
+      : 'disableAutoAwesome';
+    chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
+      chrome.tabs.sendMessage(tabs[0].id, {
+        autoAwesome: currentLikeStatus,
+        message: message
+      });
+    });
+  }, [autolikeEnabled]);
+
+  const autoLikeButton = autolikeEnabled ? (
+    <input
```

Comment:
Same comments about this code as the darkmode stuff, I think there's def a way to simplify this in the future, not a concern now